etna/auto/optuna/config_sampler.py

```diff
@@ -84,10 +90,11 @@ def after_trial(self, study: Study, trial: FrozenTrial, *args, **kwargs) -> None
             current optuna study
         """
         unfinished_hashes = self._get_unfinished_hashes(study=study, current_trial=trial)
+        hash_to_sample = trial.user_attrs["hash"]
 
         if len(unfinished_hashes) == 0:
             study.stop()
-        if len(unfinished_hashes) == 1 and list(unfinished_hashes)[0] == trial.user_attrs["hash"]:
+        if len(unfinished_hashes) == 1 and list(unfinished_hashes)[0] == hash_to_sample:
```

Comment:
```suggestion
        if len(unfinished_hashes) == 1 and unfinished_hashes.pop() == hash_to_sample:
```
Since a single element is expected.